plugins/processors/scale/scale.go

```diff
@@ -0,0 +1,116 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package Scale
+
+import (
+	_ "embed"
+	"fmt"
+
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Scale) SampleConfig() string {
+	return sampleConfig
+}
+
+type Scaling struct {
+	InMin  float64  `toml:"input_minimum"`
+	InMax  float64  `toml:"input_maximum"`
+	OutMin float64  `toml:"output_minimum"`
+	OutMax float64  `toml:"output_maximum"`
+	Fields []string `toml:"fields"`
+
+	factor      float64
+	fieldFilter filter.Filter
+}
+
+type Scale struct {
+	Scalings []Scaling       `toml:"scaling"`
+	Log      telegraf.Logger `toml:"-"`
+}
+
+func (s *Scaling) Init() error {
+	scalingFilter, err := filter.Compile(s.Fields)
+
+	if err != nil {
+		return fmt.Errorf("could not compile filter: %w", err)
+	}
+
+	s.fieldFilter = scalingFilter
```

Comment:
Please use empty lines wisely as this clutters code otherwise and harms readability. Try to build "functional blocks" like
```suggestion
	scalingFilter, err := filter.Compile(s.Fields)
	if err != nil {
		return fmt.Errorf("could not compile filter: %w", err)
	}
	s.fieldFilter = scalingFilter
```